NossoMercadoLivre/Models/Entities/Category.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Dapper.Contrib.Extensions;
+using System.ComponentModel.DataAnnotations;
+
+namespace NossoMercadoLivre.Models.Entities
+{
+    [Table("Categories")]
+    public class Category
+    {
+        [Dapper.Contrib.Extensions.Key]
+        public long Id { get; private set; }
```

Comment:
pq temos um private set aqui? Outra pergunta.. Pq tem o id é um tipo primitivo? Pergunto isso pq o valor default para o um tipo primitivo é zero.. por outro lado o objeto nasce sem id, ou seja com nulo. Se você mantiver assim, em algum momento, vai começar a brotar um != 0 no código, o que é um código meio mágico.

Reply:
- o ORM precisa do set para setar o Id na persistência, dai, deixei privado, para ninguém mais utilizar.
- No c# o "long" é um "Int64", assim como "int" é um "Int32", o valor default sempre vai ser zero, para eu aceitar null, eu teria que fazer "long?", no caso o que você quer me dizer é que é preferível eu fazer != null, ao invés de != 0 ?

Reply:
Opa, em relação ao uso da api, o melhor é ter uma abstração que represente certinho a possibilidade de um retorno nulo. Tipo o Optional do java, quando for necessário trabalhar com um valor que talvez não exista. 

Mas sim, nesse caso aí eu usaria o long? . Já que antes do id ser setado ele é nulo.